File: backend/MealPrepper.Core/Entities/ShoppingList.cs
namespace MealPrepper.Core.Entities;

public class ShoppingList
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public DateTime FromDate { get; set; }
    public DateTime ToDate { get; set; }
    public DateTime GeneratedAt { get; set; }
    public User User { get; set; } = null!;
    public ICollection<ShoppingListItem> Items { get; set; } = [];
}

File: frontend/src/types/shoppingList.ts
export interface ShoppingListItem {
  id: string
  foodId: string
  foodName: string
  unit: string
  totalQuantity: number
  isChecked: boolean
}

export interface ShoppingList {
  id: string
  userId: string
  fromDate: string
  toDate: string
  generatedAt: string
  items: ShoppingListItem[]
}
